src/app/shared/services/auth-guard.service.ts

```diff
@@ -0,0 +1,19 @@
+import {Injectable} from '@angular/core';
+import {ActivatedRouteSnapshot, CanActivate, Router, RouterStateSnapshot} from "@angular/router";
+import {AuthService} from "./auth.service";
+
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AuthGuard implements CanActivate {
+
+  constructor(private authService: AuthService, private router: Router) {
+  }
+
+  canActivate(route: ActivatedRouteSnapshot, state: RouterStateSnapshot) {
+   if (this.authService.isLoggedIn()) return true;
+   this.router.navigate(['/login'],{queryParams:{returnUrl:state.url}});
+   return false;
```

Comment:
```suggestion
  return  this.router.createUrlTree(['/login'],{queryParams:{returnUrl:state.url}});
```

instead of returning false and manually navigate the user
you should instead return the URL directly, it's more consistent with the angular framework